### DISCUSSION TITLE
useEffect not watching changings

### DISCUSSION THREAD

**Discussion by tigpalmas** [rrousselGit/flutter_hooks#269] [Q&A]
im new in flutter and trying to watch a variable "count "that is in state of reducer like React native. I made my redux and hook work perfeclty. The variable count change in the screen but never call useEffect again(only once) event if i change de action. Tried diferente aproachs without success and tried to put StoreProvider.of(context).state.balanceState.count directly in the dependecis of useEffect. If i use useState works like should.

class StatePage extends HookWidget {
  const StatePage({Key key}) : super(key: key);   

  @override
  Widget build(BuildContext context) {

    int count = StoreProvider.of<AppState>(context).state.balanceState.count;

    useEffect(() {

      WidgetsBinding.instance.addPostFrameCallback((_) =>
          StoreProvider.of<AppState>(context).dispatch(getBalanceThunk()));
    }, [count]);

    return Scaffold(
      appBar: AppBar(
        title: Text('ola'),
      ),
      body: Center(
        child: StoreConnector<AppState, BalanceState>(
          converter: (store) => store.state.balanceState,
          builder: (context, state) {
            return Column(
              mainAxisAlignment: MainAxisAlignment.center,
              crossAxisAlignment: CrossAxisAlignment.center,
              children: [
                if (state.loading)
                  Column(
                    children: [
                      CircularProgressIndicator(),
                      SizedBox(
                        height: 30,
                      )
                    ],
                  ),
                Text(state.message),
                Text(count.toString()),
                RaisedButton(
                    child: Text('Clicar'),
                    onPressed: (){
                      StoreProvider.of<AppState>(context).dispatch(SetCountAction(count++));
                    })
              ],
            );
          },
        ),
      ),
    );
  }
}

**Comment by rrousselGit** [comment]
Make sure that your widget is properly rebuilding. Try adding a print before the `useEffect` to see if it did rebuild
It's probably that your `StoreProvider.of(...)` did not cause the widget to rebuild when the count changes, hence the issue.
